tools/src/main/scala/scala/scalanative/build/Config.scala

```diff
@@ -15,16 +15,47 @@ sealed trait Config {
   def testConfig: Boolean
 
   /** Directory to emit intermediate compilation results. Calculated based on
-   *  [[basedir]] / native or native-test if a test project. The build creates
-   *  directories if they do not exist.
+   *  [[#basedir]] / `native` or `native-test` if a test project. The build
+   *  creates directories if they do not exist.
    */
   def workdir: Path
 
-  /** Path to the output file, executable or library. Calculated based on
-   *  [[basedir]] / [[NativeConfig#basename]] and -test, if a test project.
+  /** Name of the project module from the build system. Must be unique amongst
+   *  modules in the larger project.
+   *
+   *  @return
+   *    moduleName
+   */
+  def moduleName: String
+
+  /** Base name for executable or library, typically the project/module name
+   *  from the build tool [[#moduleName]] or can be overridden by the user with
+   *  [[NativeConfig#basename]]. This must be unique over all module names and
+   *  other `basename`s in the project. Delegated method to
+   *  [[NativeConfig#basename]]
+   */
+  def basename: String
```

Comment:
Both `basedir` and `basename` are like like that perhaps because of the UNIX command of the same name. I can change both of those to toggle. It would be more consistent.